packages/esm-service-queues-app/src/queue-rooms/queue-room-form.workspace.tsx

```diff
@@ -1,164 +1,170 @@
 import { saveQueueRoom } from './queue-room.resource';
 import styles from './queue-room-form.scss';
 import { useQueues } from '../hooks/useQueues';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Controller, useForm } from 'react-hook-form';
+import type { TFunction } from 'react-i18next';
+
+const createQueueRoomSchema = (t: TFunction) =>
+  z.object({
+    queueRoomName: z
+      .string({
+        required_error: t('missingQueueRoomName', 'Missing queue room name'),
+      })
+      .trim()
+      .min(1, t('missingQueueRoomName', 'Missing queue room name')),
+    queueRoomService: z
+      .string({
+        required_error: t('missingQueueRoomService', 'Missing queue room service'),
+      })
+      .trim()
+      .min(1, t('missingQueueRoomService', 'Missing queue room service')),
+    selectedQueueLocation: z
+      .string({
+        required_error: t('queueLocationRequired', 'Queue location is required'),
+      })
+      .trim()
+      .min(1, t('queueLocationRequired', 'Queue location is required')),
+  });
+
+type QueueRoomFormData = z.infer<ReturnType<typeof createQueueRoomSchema>>;
 
 const QueueRoomForm: React.FC<DefaultWorkspaceProps> = ({ closeWorkspace }) => {
   const { t } = useTranslation();
   const isTablet = useLayoutType() === 'tablet';
-  const [queueRoomName, setQueueRoomName] = useState('');
-  const [queueRoomService, setQueueRoomService] = useState('');
-  const [isMissingRoomName, setMissingRoomName] = useState(false);
-  const [isMissingQueueRoomService, setMissingQueueRoomService] = useState(false);
-  const [selectedQueueLocation, setSelectedQueueLocation] = useState('');
+  const {
+    control,
+    handleSubmit,
+    watch,
+    formState: { errors, isSubmitting },
+  } = useForm<QueueRoomFormData>({
+    resolver: zodResolver(createQueueRoomSchema(t)),
+    defaultValues: {
+      queueRoomName: '',
+      queueRoomService: '',
+      selectedQueueLocation: '',
+    },
+  });
+
+  const selectedQueueLocation = watch('selectedQueueLocation');
   const { queues } = useQueues(selectedQueueLocation);
   const { queueLocations } = useQueueLocations();
 
-  const createQueueRoom = useCallback(
-    (event) => {
-      event.preventDefault();
+  const onSubmit = async (data: QueueRoomFormData) => {
+    try {
+      // here doubt why queueName twice // name: string, description: string, queueUuid: string
+      const response = await saveQueueRoom(data.queueRoomName, data.queueRoomName, data.queueRoomService);
 
-      if (!queueRoomName) {
-        setMissingRoomName(true);
-        return;
+      if (response.status === 201) {
+        showSnackbar({
+          title: t('addQueueRoom', 'Add queue room'),
+          kind: 'success',
+          subtitle: t('queueRoomAddedSuccessfully', 'Queue room added successfully'),
+        });
+        closeWorkspace();
+        mutate(`${restBaseUrl}/queueroom`);
```

Comment:
```suggestion
        await mutate(`${restBaseUrl}/queueroom`);
```

`mutate` is a promise.